core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -467,42 +471,79 @@ private void processProviderRequest(TransferProcess process, DataRequest dataReq
         }
     }
 
+    private boolean processConsumerRequest(TransferProcess process, DataRequest dataRequest) {
+        int retryCount = process.getStateCount() - 1;
+        if (retryCount > 0) {
+            var delayStrategy = new ExponentialWaitStrategy(sendRetryBaseDelay);
+            delayStrategy.failures(retryCount);
+            var waitMillis = delayStrategy.retryInMillis();
+            long remainingWaitMillis = process.getStateTimestamp() + waitMillis - clock.millis();
+            if (remainingWaitMillis > 0) {
+                monitor.debug(format("Process %s transfer retry #%d will not be attempted before %d ms.", process.getId(), retryCount, remainingWaitMillis));
+
+                // Break lease
+                transferProcessStore.update(process);
+
+                return false;
+            }
+            monitor.debug(format("Process %s transfer retry #%d of %d.", process.getId(), retryCount, sendRetryLimit));
+        }
+        sendConsumerRequest(process, dataRequest);
+        return true;
+    }
+
     private void sendConsumerRequest(TransferProcess process, DataRequest dataRequest) {
+        String id = process.getId();
+        monitor.debug(format("TransferProcessManager: Sending process %s request to %s", process.getId(), dataRequest.getConnectorAddress()));
         dispatcherRegistry.send(Object.class, dataRequest, process::getId)
                 .thenApply(result -> {
-                    var transferProcess = transferProcessStore.find(process.getId());
-
-                    if (transferProcess == null) {
-                        monitor.severe(format("TransferProcessManager: no TransferProcess found with id %s", process.getId()));
-                        throw new EdcException(format("TransferProcess %s not found", process.getId()));
-                    }
-
-                    transferProcess.transitionRequested();
-                    transferProcessStore.update(transferProcess);
-                    observable.invokeForEach(l -> l.preRequested(transferProcess));
+                    sendConsumerRequestSuccess(id);
```

Comment:
nit: maybe its better to have `TransferProcess` as param here instead of `id` as then we do not need to fetch  `TransferProcess` twice , once in `sendConsumerRequestSuccess` and once in `sendCustomerRequestFailure`.

Or do you think that would cause some other issue ?

Reply:
Thanks, it seemed reasonable to refresh the entity after an async operation, but I see we're not even checking that we still have the lease, so in this form it brings no value. Changed per your suggestion.